### ISSUE TITLE
Convert uses of `null` to not adding fields

### ISSUE THREAD

**Issue by patricoferris** [geocaml/ocaml-geojson#28]
At the moment, if an optional argument is converted to JSON we use `Option.(value ~default:J.null props)` which means we still output the empty fields (which is technically okay, but it is extra bytes for our end users). 

```ocaml
utop # Ezjsonm.to_string (`O [ "hello", `Null  ]);;
- : string = "{\"hello\":null}"
``` 

I think we should use `Option.to_list` and concatenate the fields together instead so we don't output `` `Null`` values. This is similar to this function https://github.com/geocaml/ocaml-geojson/commit/fb140881a6e16390be5070aa018a1f23e74c4bc9#diff-039fc6a826ee8bf3666c5ae99c2f249f84272ac63a754285a0132236296d53ceR26-R29.

The places that need changed are places like https://github.com/geocaml/ocaml-geojson/blob/27ffcf9f7de40cd73295289fb5d3451ad628b75f/src/geojson/geojson.ml#L303

@SHIVANGISINGH1 would you like to work on this? :))

**Comment by imajinarts** [comment]
@techbae22 here's one free issue you could ask permission from our mentor to work on

**Comment by patricoferris** [comment]
Thanks for the help @imajinarts :))

Please note I've asked @SHIVANGISINGH1 if they would like to work on this first, if not @Techbae22 feel free to register your interest, I'll also think of more issues too.

**Comment by Techbae22** [comment]
Hello @patricoferris, 
I had registered my interest to contribute to issue: https://github.com/geocaml/ocaml-geojson/issues/26 and it doesn't look like it has been assigned to anyone yet. However, it seems this issue was created with @SHIVANGISINGH1 in mind. Can you consider having a new issue assigned to me ?

**Comment by patricoferris** [comment]
Ah thanks for reminding me @Techbae22, I have assigned you to #26

**Comment by SHIVANGISINGH1** [comment]
> Thanks for the help @imajinarts :))
> 
> Please note I've asked @SHIVANGISINGH1 if they would like to work on this first, if not @Techbae22 feel free to register your interest, I'll also think of more issues too.

Yes, I would like to work on this issue. Thanks, @patricoferris for considering me 🙇‍♀️ 
I will start working on this and will ask my doubts here. 😃

**Comment by Techbae22** [comment]
> Ah thanks for reminding me @Techbae22, I have assigned you to #26

Great ! Thank you @patricoferris

**Comment by SHIVANGISINGH1** [comment]
Hello @patricoferris I just had some doubts regarding this issue

Here `Option.(value ~default:J.null props)`  is returning an empty null object as default otherwise giving properties object, so we don’t want to display the null object in both the cases ? 

And it is a little unclear to me how it is similar to this function https://github.com/geocaml/ocaml-geojson/commit/fb140881a6e16390be5070aa018a1f23e74c4bc9#diff039fc6a826ee8bf3666c5ae99c2f249f84272ac63a754285a0132236296d53ceR26-R29. as it returns an empty array. 

Please guide me with this.🙏

**Comment by SHIVANGISINGH1** [comment]
@streetCoderr would you like to help me with this  

> Hello @patricoferris I just had some doubts regarding this issue
> 
> Here `Option.(value ~default:J.null props)` is returning an empty null object as default otherwise giving properties object, so we don’t want to display the null object in both the cases ?
> 
> And it is a little unclear to me how it is similar to this function [fb14088#diff039fc6a826ee8bf3666c5ae99c2f249f84272ac63a754285a0132236296d53ceR26-R29](https://github.com/geocaml/ocaml-geojson/commit/fb140881a6e16390be5070aa018a1f23e74c4bc9#diff039fc6a826ee8bf3666c5ae99c2f249f84272ac63a754285a0132236296d53ceR26-R29). as it returns an empty array.
> 
> Please guide me with this.🙏

@streetCoderr would you like to help me with this till @patricoferris responds, as you worked with the same file?

**Comment by patricoferris** [comment]
> And it is a little unclear to me how it is similar to this function https://github.com/geocaml/ocaml-geojson/commit/fb140881a6e16390be5070aa018a1f23e74c4bc9#diff039fc6a826ee8bf3666c5ae99c2f249f84272ac63a754285a0132236296d53ceR26-R29. as it returns an empty array.

In the representation of JSON that we are using, a JSON object is represented as a list of key-value pairs (`(string * json) list`). And so we end up with two ways of saying that a value is not there either we put the key in and the value is `null` or we simply leave the key-value pair out altogether. For example:

```ocaml
[ "name", J.string "Alice; "age", J.null]
```

Or

```ocaml
[ "name", J.string "Alice"] @ []
```

Where the empty list is generated because there was no age to put. Does that help a little?

**Comment by streetCoderr** [comment]
Thanks for the mention @SHIVANGISINGH1. 
Of course, I would definitely love to help.

**Comment by Bash-dot** [comment]
Hi sir @patricoferris  can I please have this issue assigned to me.

**Comment by SHIVANGISINGH1** [comment]
> How is this going @SHIVANGISINGH1 ? No worries if you've decided to not work on it, if you could just let us know that would be great. Otherwise, if you need any help do let me know as well.

Hello, @patricoferris I am sorry but unfortunately, I am not working on this issue anymore.

**Comment by patricoferris** [comment]
No worries at all, @Bash-dot feel free to work on this.

**Comment by patricoferris** [comment]
Fixed in #21
